Upgrading install script to install & prepare a Postgres + DfE Sign-in.

NyWeb opened the pull request:
### Context
Getting both backend and frontend up and running took me (Rails novice) 30min, so the instructions are good. I ran into issues with Postgres, and DfE sign-in seems to be a requirement.

##### DfE Sign-in
Technically DfE Sign-in is needed in front-end, not in this library, but the commands are run on mcb, hence it being here.

Traditionally a god-user created another god-user. That gives the new user access to all current providers - with the awkward side-effect of an inverse relationship between the number of providers on your profile and the amount of time you've been a dev for Find. Here I humbly guess that adding specific providers directly to your user might be the way to go for future devs.

### Changes proposed in this pull request
Adding a chapter on DfE sign-in, created with @timabell's help. Also added a user-creation command to get past a macOS specific bug.

### Guidance to review
First PR on this project, to give feedback

### Checklist
- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

NyWeb commented:
![](https://github.trello.services/images/mini-trello-icon.png) [README update for macOS + DfE Sign-in](https://trello.com/c/GWEssvOl/1786-readme-update-for-macos-dfe-sign-in)

defong commented on README.md:
```diff
@@ -23,6 +23,7 @@
 
 1. Run `bundle install` to install the gem dependencies
 2. Run `bundle exec rails db:setup` to create a development and testing database
+   1. MacOS: If you get an error `FATAL: role "postgres" does not exist` run `createuser -s postgres` in your terminal
```
I think this applies to postgres regard less of OS

@timabell for input

NyWeb replied:
Ok, MacOS removed :)

defong commented on README.md:
```diff
@@ -191,3 +192,21 @@ be organised in an appropriate sub-folder there.
+
+### DfE Sign-in
+
+To use it locally (and in qa) you will need a DfE Sign-in account.
```
maybe reword it to state along the veins of `mandatory requirements` 
ie
to successfully login to system
1.  need to create an account on dfe
2.  assign organisation/provider or via god user

Otherwise you stuck

NyWeb replied:
Thanks, much clearer now!

defong commented on README.md:
```diff
@@ -191,3 +192,21 @@ be organised in an appropriate sub-folder there.
+https://signin-test-sup-as.azurewebsites.net/users)
+2. Sign up from the email sent from DfE Sign-in
+3. Grant access to providers (here on production): 
+`bundle exec bin/mcb users -E production grant {email} {course_code}`
```
mismatch

bullet point 1 uses tests url
bullet point 3 uses production

NyWeb replied:
Added context

defong commented on README.md:
```diff
@@ -191,3 +192,21 @@ be organised in an appropriate sub-folder there.
+3. Grant access to providers (here on production): 
+`bundle exec bin/mcb users -E production grant {email} {course_code}`
+
+Note that all users are anonymized and synced to `qa` daily,
```
reword it the other way please, all other environments uses anonymized not just `qa` or `localhost`

NyWeb replied:
How about "Users not matching `%@digital.education.gov.uk` and `%@education.gov.uk` will be anonymized."?

defong commented on README.md:
```diff
@@ -47,7 +48,7 @@ docker-compose exec web /bin/sh -c "bundle exec rails db:setup"
 Then open http://localhost:3001 to see the app.
 
-## Running specs, linter(without auto correct) and annotate models and serializers
+## Running specs, linter (without auto correct) and annotate models and serializers
```
I think its with auto correct

@dankmitchell for input

NyWeb replied:
It's more legible and correct I believe.

defong replied:
opps I meant this bit

linter (without auto correct)
vs
linter (with auto correct)

not the spacing linter(with auto correct)

defong commented on README.md:
```diff
@@ -191,3 +192,19 @@ be organised in an appropriate sub-folder there.
+   2. Sign up from the email sent from DfE Sign-in
+2. Grant access to some providers: 
+   1. For local: `bundle exec bin/mcb users grant {email} {course_code}`
+   2. For `qa` and `production`: `bundle exec bin/mcb -E production users grant {email} {course_code}`
```
prefer correct ordering

```
2. For `qa` and `production`:
  1. You will need to log into Azure first: `az login`
  2. `bundle exec bin/mcb -E {env} users grant {email} {provider_code}`
```
should be  {provider_code} vs {course_code}

NyWeb replied:
Apologies, what does `should be {provider_code} vs {course_code}` mean?

defong replied:
didn't check the command...

but grant a user access to a provider I would think it needs provider code vs the course code
https://github.com/DFE-Digital/manage-courses-backend/blob/f517332bb30864e57946da82f940e68feb10e049/lib/mcb/commands/users/grant.rb#L4

so i guess command should be something like, 
`
`bundle exec bin/mcb -E {env} grant {email} {provider_code}`
`

defong commented on README.md:
```diff
@@ -191,3 +192,19 @@ be organised in an appropriate sub-folder there.
+
+Users not matching
+`%@digital.education.gov.uk` and `%@education.gov.uk`
+will be anonymized.
```
```suggestion
will be anonymized for non production environment.
```

NyWeb commented:
After a conversation with @dankmitchell and @misaka I renamed and updated `lib/tasks/create_dev_user.rake` to install Postgres and fix the user issue we've had. I also restructured that script quite a bit - hopefully for the better, but I left one switch clause too many perhaps.
